libnavui-tripprogress/src/main/java/com/mapbox/navigation/ui/tripprogress/TripProgressProcessor.kt

```diff
@@ -2,12 +2,53 @@ package com.mapbox.navigation.ui.tripprogress
 
 import com.mapbox.navigation.base.trip.model.RouteProgress
 import com.mapbox.navigation.base.trip.model.RouteProgressState
+import com.mapbox.navigation.utils.internal.ifNonNull
 import java.util.Calendar
 
 internal class TripProgressProcessor {
 
     fun process(action: TripProgressAction): TripProgressResult {
-        return calculateTripProgress(action as TripProgressAction.CalculateTripProgress)
+        return when (action) {
+            is TripProgressAction.CalculateTripDetails -> {
+                calculateTripDetails(action)
+            }
+            is TripProgressAction.CalculateTripProgress -> {
+                calculateTripProgress(action)
+            }
+        }
+    }
+
+    private fun calculateTripDetails(
+        action: TripProgressAction.CalculateTripDetails
+    ): TripProgressResult.TripDetails {
+        val tripDetailsList = mutableListOf<TripProgressResult.RouteLegTripDetail>()
+        ifNonNull(action.route.directionsRoute.legs()) { legs ->
+            legs.forEach { routeLeg ->
+                val eta = routeLeg.duration()?.let { duration ->
+                    Calendar.getInstance().also {
+                        it.add(Calendar.SECOND, duration.toInt())
+                    }.timeInMillis
+                } ?: 0L
+                val legTime = routeLeg.duration() ?: 0.0
+                val legDistance = routeLeg.distance() ?: 0.0
```

Comment:
return an error instead

Reply:
Do we want to throw an error? Why not make 
```
data class RouteLegTripDetail(
        val legTime: Double?,
        val legDistance: Double?,
        val estimatedTimeToArrival: Long?,
    )
```
the properties nullable?

@LukasPaczos

Reply:
Or, better yet what do you think about this:

```
ifNonNull(action.route.directionsRoute.legs()) { legs ->
            legs.forEach { routeLeg ->
                ifNonNull(routeLeg.duration(), routeLeg.distance()) { duration, distance ->
                    val eta = Calendar.getInstance().also {
                        it.add(Calendar.SECOND, duration.toInt())
                    }.timeInMillis
                    tripDetailsList.add(
                        TripProgressResult.RouteLegTripOverview(
                            routeLeg = routeLeg,
                            estimatedTimeToArrival = eta,
                            legTime = duration,
                            legDistance = distance,
                        )
                    )
                }
            }
        }
```
In this case the list is only prepared if `RouteLeg` is not null along with `RouteLeg.duration` and `RouteLeg.distance` are not null as well. We can capture in the documentation that the API will return an `emptyList()` if any of the above is null

Reply:
Not seeing any use case where one would ever get an error. Similar to `render()` based on `RouteProgress`. It always has a value.

Reply:
You already checking for that case in `ifNonNull(action.route.directionsRoute.legs())`.

Reply:
added `Expected<SomeError, TripDetailsValue>`. @LukasPaczos